src/comodo/jaxsimSimulator/jaxsimSimulator.py

```diff
@@ -0,0 +1,465 @@
+
+        # Optional Mujoco model helper
+        self._mj_model_helper: Optional[MujocoModelHelper] = None
+
+    # ==== Simulator interface methods ====
+
+    def load_model(
+        self,
+        robot_model: RobotModel,
+        *,
+        dt: float = 0.001,
+        xyz_rpy: npt.ArrayLike = np.zeros(6),
+        contact_model_type: ContactModelEnum = ContactModelEnum.RELAXED_RIGID,
+        s: Optional[npt.ArrayLike] = None,
+        contact_params: Optional[ContactsParams] = None,
+        left_foot_link_name: Optional[str] = "l_ankle_2",
+        right_foot_link_name: Optional[str] = "r_ankle_2",
+        left_foot_sole_frame_name: Optional[str] = "l_sole",
+        right_foot_sole_frame_name: Optional[str] = "r_sole",
+        visualization_mode: Optional[str] = None,
+    ) -> None:
+        # ==== Initialize simulator model and data ====
+
+        self._dt = dt
+        self._contact_model_type = contact_model_type
+
+        match contact_model_type:
+            case ContactModelEnum.RIGID:
+                contact_model = RigidContacts()
+            case ContactModelEnum.RELAXED_RIGID:
+                contact_model = RelaxedRigidContacts()
+            case ContactModelEnum.VISCO_ELASTIC:
+                contact_model = ViscoElasticContacts()
+            case ContactModelEnum.SOFT:
+                contact_model = SoftContacts()
+            case _:
+                raise ValueError(f"Invalid contact model type: {contact_model_type}")
+
+        model = js.model.JaxSimModel.build_from_model_description(
+            model_description=robot_model.urdf_string,
+            model_name=robot_model.robot_name,
+            contact_model=contact_model,
+        )
+
+        model = js.model.reduce(
+            model=model,
+            considered_joints=robot_model.joint_name_list,
+        )
+
+        self._model = model
+
+        if contact_params is None:
+            match contact_model_type:
+                case ContactModelEnum.RIGID:
+                    contact_params = RigidContactsParams.build(mu=0.5, K=1.0e4, D=1.0e2)
+                case ContactModelEnum.RELAXED_RIGID:
+                    contact_params = RelaxedRigidContactsParams.build(
+                        mu=0.8, time_constant=0.005
+                    )
+                case ContactModelEnum.VISCO_ELASTIC | ContactModelEnum.SOFT:
+                    contact_params = js.contact.estimate_good_soft_contacts_parameters(
+                        model=self._model,
+                        number_of_active_collidable_points_steady_state=16,
+                        max_penetration=0.002,
+                        damping_ratio=1.0,
+                        static_friction_coefficient=1.0,
+                    )
+                case _:
+                    raise ValueError(
+                        f"Invalid contact model type: {contact_model_type}"
+                    )
+
+        s = s or np.zeros(self._model.dofs())
```

Comment:
Is this necessary? If I recall, `JaxSimModelData.build` should already take care to do this if `joint_positions=None`.

Reply:
At the end I updated that statement since there now i also sort the joint positions for jaxsim

Reply:
Done in 92f3d5e703956a1e4c3330393bccd1e58497cf7d